### PR TITLE
Add restore database script

### PR DESCRIPTION
**Pull request by MancunianSam**
- Initial commit for db restore spike
- Add script and README to restore DB from snapshot

### CODE REVIEW

File: terraform/restore-database/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export TF_VAR_tdr_account_number=xxxxxxxxxxx
+export DB_NO_DASH=$(echo $TF_VAR_database | sed  's/-//g')
+export TF_VAR_cluster_identifier=$(aws rds describe-db-clusters --query="DBClusters[?DatabaseName=='$DB_NO_DASH'].DBClusterIdentifier" --output text)
+export PROFILE=intg # replace with staging or prod if using those environments
+
+terraform init
+terraform apply
+```
+
+The restore may take 10-15 minutes depending on the size of the database when you run this.
```
Comment (suzannehamilton) [inline terraform/restore-database/README.md:19]: How do you know when it is finished? Does the Terraform script not complete until the restore has finished, or do you have to check on the console?

File: terraform/restore-database/root.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  final_snapshot_identifier           = "restored-${var.database}-final-snapshot-${random_string.identifier.result}-${local.environment}"
+  tags                                = local.common_tags
+}
+
+resource "aws_rds_cluster_instance" "atabase_instance" {
+  count                = 1
+  identifier_prefix    = "db-postgres-instance-${local.environment}"
+  cluster_identifier   = aws_rds_cluster.db_restore_cluster.id
+  engine               = "aurora-postgresql"
+  engine_version       = "11.9"
```
Comment (suzannehamilton) [inline terraform/restore-database/root.tf:35]: AWS upgrades the DBs periodically. Will it cause any issues if this is older than the prod DB? Should we make it a parameter, or make it match the prod DB version by reading it as a data source?

**Review by suzannehamilton — COMMENTED** [review]
Just a few small comments, thanks!

File: terraform/restore-database/README.md (hunk 1 above)
Comment (MancunianSam) [reply terraform/restore-database/README.md:19]: Terraform will wait until the cluster is up. I've added in a line to say that.

File: terraform/restore-database/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Restore a database
+
+This script creates a replica of an existing rds database from a snapshot. It also updates the database url parameter in the parameter store to point to the new
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply terraform/restore-database/README.md:3]: Ah yeah, got sidetracked. I've finished the sentence.

[the remark this answers is not in the document]
Comment (MancunianSam) [reply terraform/restore-database/README.md:3]: Yep, they both make sense, I've added them in.

File: terraform/restore-database/root_variables.tf [hunk 4]
```diff
@@ -0,0 +1,13 @@
+variable "tdr_account_number" {}
+variable "database" {
+  description = "The database to restore. Either consignment-api or keycloak"
+}
+
+variable "restore_time" {
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply terraform/restore-database/root_variables.tf:6]: I've added in a bit about setting the environment variable and added in the command to get the earliest and latest available restore time.

File: terraform/restore-database/root.tf (hunk 2 above)
Comment (MancunianSam) [reply terraform/restore-database/root.tf:35]: It's a good point and it makes sense as a variable I think. I've added in a variable and put a bit in the README about how to get it.

**Review by suzannehamilton — APPROVED** [review]

**Pull request merged**